### PR TITLE
Fixed rare error of dropped collections in-between migrations

### PR DESCRIPTION
**Pull request by vanruch**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

**Comment by Anze1m** [comment]
So if in `...bundle...` migration we rely on non-explicit collection existence ensurement, maybe we should do the same in `worker_task_ttl`, where we call `indexExists()`?

**Comment by vanruch** [comment]
`indexExists` throws when called on not existing collection

File: src/migrations/20181206163000_move_out_bundles_metadata.js [hunk 1]
```diff
@@ -41,7 +39,8 @@ export const up = async (db, config, logger) => {
   await deduplicateBundles(db);
   await db.collection('bundle_metadata').createIndex({bundleId : 1}, {unique: true});
 
-  if (await db.collection('bundles').indexExists(['bundleId_1'])) {
+  const collectionExists = (await db.listCollections({name: 'bundles'}).toArray()).length > 0;
+  if (collectionExists && await db.collection('bundles').indexExists(['bundleId_1'])) {
```
Comment (spherefoundry) [inline src/migrations/20181206163000_move_out_bundles_metadata.js]: We have a lot of duplication here. I think it's time for some `migration_helpers.js`

File: src/migrations/20181206163000_move_out_bundles_metadata.js [hunk 2]
```diff
@@ -7,44 +7,14 @@ This Source Code Form is subject to the terms of the Mozilla Public License, v.
...
-  await deduplicateBundles(db);
-  await db.collection('bundle_metadata').createIndex({bundleId : 1}, {unique: true});
+  await deduplicateDocuments(db, 'bundles', 'bundleId');
 
-  if (await db.collection('bundles').indexExists(['bundleId_1'])) {
-    await db.collection('bundles').dropIndex('bundleId_1');
-  }
-  await db.collection('bundles').createIndex({bundleId : 1}, {unique: true});
+  await safeCreateUniqueIndex(db, 'bundle_metadata', 'bundleId');
+  await safeCreateUniqueIndex(db, 'bundles', 'bundleId');
```
Comment (spherefoundry) [inline src/migrations/20181206163000_move_out_bundles_metadata.js]: so clean 👍
